balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -56,6 +56,13 @@ func SetAddrInfo(addr resolver.Address, addrInfo AddrInfo) resolver.Address {
 	return addr
 }
 
+// SetAddrInfoInEndpoint returns a copy of endpoint in which the Attributes
+// field is updated with addrInfo.
+func SetAddrInfoInEndpoint(endpoint resolver.Endpoint, addrInfo AddrInfo) resolver.Endpoint {
+	endpoint.Attributes = endpoint.Attributes.WithValue(attributeKey{}, addrInfo)
```

Comment:
To support both addresses and endpoints during the migration, there are a couple of places that I found which convert resolver.Address to resolver.Endpoint or vice versa.
https://github.com/grpc/grpc-go/blob/44a5eb9231ec6e753dab5ded7cda6f81788fc3f7/xds/internal/balancer/clusterresolver/clusterresolver.go#L251-L255

https://github.com/grpc/grpc-go/blob/44a5eb9231ec6e753dab5ded7cda6f81788fc3f7/resolver_wrapper.go#L126-L131

If we use different keys, we would need to ensure the translation of keys is done correctly in such places. This could make the translation code a little brittle. 

Are `resolver.Attributes` a direct replacement for `Address.BalancerAttributes` or are there differences? If both can be used interchangably, I think we should use the same key.

Reply:
Yeah, makes sense to have the same key since `Endpoint.Attributes` is a direct replacement for `Address.BalancerAttributes`. Thanks.